### PR TITLE
Cleanup documents that are not referenced anywhere

### LINKED ISSUE #6848: Connectors and APIs need updating
## Change Request

https://docs.pingcap.com/tidb/stable/connectors-and-apis

Issues:
- Not referenced in the TOC
- Outdated info
- Overlapping and contradicting the info on https://docs.pingcap.com/appdev/dev/app-dev-overview

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

- Add `scripts/check-unreachable.py` to check for files that are not referenced from anywhere
- Cleanup files that seem outdated
- Add TOC links for newer documents

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

Closes #6848
Related #6623 

Some of the documents removed in this PR are replaced by:

- https://docs.pingcap.com/appdev/dev/app-dev-overview
- https://pingcap.github.io/tidb-dev-guide/

### Do your changes match any of the following descriptions?

- [x] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by dveeden** [comment]
Output from the script:
```
[dvaneeden@dve-carbon docs]$ ./scripts/check-unreachable.py 
File contribute.md has NO matches
File community.md has NO matches
File report-issue.md has NO matches
File technical-writing-project-ideas.md has NO matches
File whats-new-in-tidb-4.0.md has NO matches
File support.md has NO matches
File connectors-and-apis.md has NO matches
File dev-guide/develop.md has NO matches
File dev-guide/deploy.md has NO matches
File tidb-lightning/tidb-lightning-distributed-import.md has NO matches
```

**Comment by dveeden** [comment]
PTAL @TomShawn

File: dev-guide/deploy.md [hunk 1]
```diff
@@ -1,18 +0,0 @@
----
-title: Build for Deployment
```
Comment (TomShawn) [inline dev-guide/deploy.md:2]: All documents in the `dev-guide` directory are not parsed into web pages in docs.pingcap.com.

File: dev-guide/deploy.md [hunk 2]
```diff
@@ -1,18 +0,0 @@
...
-draft: true
----
-
-# Build for Deployment
```
Comment (TomShawn) [inline dev-guide/deploy.md:6]: @lilyjazz PTAL whether this guide can be deleted. Thanks~

File: support.md [hunk 3]
```diff
@@ -1,19 +0,0 @@
----
-title: Support Resources
-summary: Find support resources for your TiDB installation.
```
Comment (TomShawn) [inline support.md:3]: @lilin90 PTAL whether we should keep this file. It is shown in https://docs.pingcap.com/tidb/dev/support but not in the TOC menu. Thanks~

File: technical-writing-project-ideas.md [hunk 4]
```diff
@@ -1,58 +0,0 @@
----
-title: TiDB Technical Writing Project Ideas
```
Comment (TomShawn) [inline technical-writing-project-ideas.md:2]: @lilin90 PTAL, thanks~

File: whats-new-in-tidb-4.0.md [hunk 5]
```diff
@@ -1,92 +0,0 @@
----
-title: What's New in TiDB 4.0
```
Comment (TomShawn) [inline whats-new-in-tidb-4.0.md:2]: I think this file can be safely deleted after adding an aliases to https://docs.pingcap.com/tidb/v4.0/whats-new-in-tidb-4.0.

File: connectors-and-apis.md [hunk 6]
```diff
@@ -1,92 +0,0 @@
----
-title: Connectors and APIs
```
Comment (TomShawn) [inline connectors-and-apis.md:2]: This file has been there for years. Some of its contents may be outdated. We don't want users to see the entry so we remove it from TOC but still keep the file.

File: community.md [hunk 7]
```diff
@@ -1,12 +0,0 @@
----
-title: Connect with us
```
Comment (TomShawn) [inline community.md:2]: @lilin90 @CaitinChen PTAL whether we can remove this file in the docs repo. Thanks!

**Comment by TomShawn** [comment]
@dveeden If we want to remove a file, we may need confirmations because some files not in TOC or not referenced anywhere might be simply dealt with in that way on purpose.

File: support.md (hunk 3 above)
Comment (lilin90) [reply support.md:3]: @TomShawn Most of these resources are also displayed in the footer, so I agree with deleting this file. Let's focus on documentation content in `.md` files.

File: community.md (hunk 7 above)
Comment (lilin90) [reply community.md:2]: @TomShawn Yes. I think we can remove it.

File: technical-writing-project-ideas.md (hunk 4 above)
Comment (lilin90) [reply technical-writing-project-ideas.md:2]: @TomShawn We can remove this file. It's outdated and not necessary to keep it in this repo now.

File: whats-new-in-tidb-4.0.md (hunk 5 above)
Comment (lilin90) [reply whats-new-in-tidb-4.0.md:2]: @TomShawn Yes. Would you please help with adding the `aliases` to that file?

File: TOC.md [hunk 8]
```diff
@@ -203,6 +203,7 @@
       - [Backends](/tidb-lightning/tidb-lightning-backends.md)
       - [Import Data in Parallel](/tidb-lightning/tidb-lightning-distributed-import.md)
       - [Web Interface](/tidb-lightning/tidb-lightning-web-interface.md)
+      - [Import Data in Parallel](/tidb-lightning/tidb-lightning-distributed-import.md)
```
Comment (lilin90) [inline TOC.md:206]: This file has been added in line 204.
```suggestion
```

File: whats-new-in-tidb-4.0.md (hunk 5 above)
Comment (TomShawn) [reply whats-new-in-tidb-4.0.md:2]: OK

File: support.md (hunk 3 above)
Comment (TomShawn) [reply support.md:3]: Get. Thanks!

File: technical-writing-project-ideas.md (hunk 4 above)
Comment (TomShawn) [reply technical-writing-project-ideas.md:2]: Get. Thanks!

File: community.md (hunk 7 above)
Comment (TomShawn) [reply community.md:2]: Get. Thanks!

**Review by TomShawn — APPROVED** [review]
Except for Lilian's comment, rest LGTM

**Review by lilin90 — APPROVED** [review]
LGTM

**Pull request merged**